tempodb/encoding/vparquet3/schema.go

```diff
@@ -252,29 +268,96 @@ type Trace struct {
 
 func attrToParquet(a *v1.KeyValue, p *Attribute, counter droppedAttrCounter) {
 	p.Key = a.Key
-	p.Value = nil
-	p.ValueBool = nil
-	p.ValueDouble = nil
-	p.ValueInt = nil
+	p.ValueType = 0
+	p.Value = p.Value[:0]
+	p.ValueInt = p.ValueInt[:0]
+	p.ValueDouble = p.ValueDouble[:0]
+	p.ValueBool = p.ValueBool[:0]
 	p.ValueDropped = ""
 
 	switch v := a.GetValue().Value.(type) {
 	case *v1.AnyValue_StringValue:
-		p.Value = &v.StringValue
+		p.Value = append(p.Value, v.StringValue)
+		p.ValueType = attrTypeString
 	case *v1.AnyValue_IntValue:
-		p.ValueInt = &v.IntValue
+		p.ValueInt = append(p.ValueInt, v.IntValue)
+		p.ValueType = attrTypeInt
 	case *v1.AnyValue_DoubleValue:
-		p.ValueDouble = &v.DoubleValue
+		p.ValueDouble = append(p.ValueDouble, v.DoubleValue)
+		p.ValueType = attrTypeDouble
 	case *v1.AnyValue_BoolValue:
-		p.ValueBool = &v.BoolValue
+		p.ValueBool = append(p.ValueBool, v.BoolValue)
+		p.ValueType = attrTypeBool
+	case *v1.AnyValue_ArrayValue:
+		if v.ArrayValue == nil || len(v.ArrayValue.Values) == 0 {
+			p.ValueType = attrTypeStringArray
+			return
+		}
+		switch v.ArrayValue.Values[0].Value.(type) {
+		case *v1.AnyValue_StringValue:
+			for _, e := range v.ArrayValue.Values {
+				ev, ok := e.Value.(*v1.AnyValue_StringValue)
+				if !ok {
+					p.Value = p.Value[:0]
+					attrToParquetTypeUnsupported(a, p, counter)
+					return
+				}
+
+				p.Value = append(p.Value, ev.StringValue)
+				p.ValueType = attrTypeStringArray
+			}
+		case *v1.AnyValue_IntValue:
+			for _, e := range v.ArrayValue.Values {
+				ev, ok := e.Value.(*v1.AnyValue_IntValue)
+				if !ok {
+					p.ValueInt = p.ValueInt[:0]
+					attrToParquetTypeUnsupported(a, p, counter)
+					return
+				}
+
+				p.ValueInt = append(p.ValueInt, ev.IntValue)
+				p.ValueType = attrTypeIntArray
+			}
+		case *v1.AnyValue_DoubleValue:
+			for _, e := range v.ArrayValue.Values {
+				ev, ok := e.Value.(*v1.AnyValue_DoubleValue)
+				if !ok {
+					p.ValueDouble = p.ValueDouble[:0]
+					attrToParquetTypeUnsupported(a, p, counter)
+					return
+				}
+
+				p.ValueDouble = append(p.ValueDouble, ev.DoubleValue)
+				p.ValueType = attrTypeDoubleArray
+			}
+		case *v1.AnyValue_BoolValue:
+			for _, e := range v.ArrayValue.Values {
+				ev, ok := e.Value.(*v1.AnyValue_BoolValue)
+				if !ok {
+					p.ValueBool = p.ValueBool[:0]
+					attrToParquetTypeUnsupported(a, p, counter)
+					return
+				}
+
+				p.ValueBool = append(p.ValueBool, ev.BoolValue)
+				p.ValueType = attrTypeBoolArray
+			}
+		default:
+			attrToParquetTypeUnsupported(a, p, counter)
+		}
 	default:
-		jsonBytes := &bytes.Buffer{}
-		_ = jsonMarshaler.Marshal(jsonBytes, a.Value) // deliberately marshalling a.Value because of AnyValue logic
-		p.ValueDropped = jsonBytes.String()
-		counter.addDroppedAttr(1)
+		attrToParquetTypeUnsupported(a, p, counter)
 	}
 }
 
+func attrToParquetTypeUnsupported(a *v1.KeyValue, p *Attribute, counter droppedAttrCounter) {
```

Comment:
yeah, this is probably ok. most of our issues have been with dictionary sizes which this table is not. i think ideally we'd have a way to disable this behavior in case someone found a way to panic ingesters with it, but my guess is it won't come up.